src/Plugin/Validators/ValidDelimitedFile.php

```diff
@@ -135,10 +190,168 @@ public function validateRawRow(string $raw_row) {
+   *     - 'strict-or-min': header comparison settings (strict or minimum).
+   *     - 'num-expected-colums': number of expected column headers.
+   *
+   * @return array
+   *   A render array of type unordered list which is used to display feedback
+   *   to the user about the case(s) that failed and the failed items from the
+   *   input file. This unordered list will include a table for each potential
+   *   case in the $failures array:
+   *   - A table for lines that are empty or contain unsupported delimiters
+   *   - A table for lines that once delimited, do not contain the expected
+   *     number of columns.
+   *   Both tables contain the following headers:
+   *   - 'Line Number'
+   *   - 'Line Contents'
+   *
+   * @throws \Exception
+   *   - If the validation_results parameter was not formatted properly.
+   *   - If the case string returned by the validator implied validation passed.
+   *   - If the case string returned by the validator is not recognized.
+   */
+  public static function processValidDelimitedFileFailures(array $validation_results, array $tokens = []) {
+
+    // Grab the default messages for all of our tokens (ones with default-msg).
+    $default_tokens = array_column(self::$mapping, 'default-msg', 'token');
+    // Combine our provided and our default token arrays. Because array_merge
+    // will overwrite values in the first array with values from the second
+    // array for the same keys, we provide our default tokens first.
+    $combined_tokens = array_merge($default_tokens, $tokens);
+
+    // Define our table headers.
+    $table_header = ['Line Number', 'Line Contents'];
+
+    // For this validator there can be up to 2 tables:
+    // - 'table'->'unsupported': Empty rows or no supported delimiters present.
+    // - 'table'->'delimited': Rows that don't delimit to the expected number of
+    //   columns.
+    $table = [];
+    // Loop through each row in the $failures array and piece apart the
+    // different cases into different tables.
+    foreach ($validation_results as $line_no => $validation_result) {
+      // Check the format of the validation_result parameter.
+      ImportValidationHelper::checkValidationStatusArray($validation_result, 'ValidDelimitedFile', $line_no);
+      // Keeps track of which table this one line's validation result gets added
+      // to based on the case it triggered.
+      $table_case = '';
+      if (($validation_result['case'] == self::$mapping['case-empty-row']['dev-case']) ||
+          ($validation_result['case'] == self::$mapping['case-no-delimiter']['dev-case'])) {
+
+        $table_case = 'unsupported';
+      }
+      elseif (($validation_result['case'] == self::$mapping['case-excess-columns']['dev-case']) ||
+              ($validation_result['case'] == self::$mapping['case-insufficient-columns']['dev-case'])) {
+
+        $table_case = 'delimited';
+        if (!isset($num_expected_columns)) {
```

Comment:
This is the only time this variable is seen, so it will never be set. I think it should be safe to remove this if completely, since the 2 cases leading here should have `expected_columns` and `strict` in their failedItems array already. The only problem is that these values in $combined_tokens will be set over and over if a large file with a lot of wrong number of columns is validated.